pkg/broker/ingress/multi_topic_decouple_sink_test.go

```diff
@@ -220,6 +226,219 @@ func TestMultiTopicDecoupleSink(t *testing.T) {
 	}
 }
 
+func TestHasTrigger(t *testing.T) {
+	DecoupleQueue := &config.Queue{Topic: "test_topic", State: config.State_READY}
+
+	tests := []struct {
+		name          string
+		brokerTargets map[string]*config.Target
+		hasTrigger    bool
+	}{
+		{
+			name:       "broker with no target",
+			hasTrigger: false,
+		},
+		{
+			name: "broker with target with no filters",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {},
+			},
+			hasTrigger: true,
+		},
+		{
+			name: "broker with target with matching filter",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "test-source",
+					},
+				},
+			},
+			hasTrigger: true,
+		},
+		{
+			name: "broker with target with non-matching filter",
+			brokerTargets: map[string]*config.Target{
+				"target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "some-random-source",
+					},
+				},
+			},
+			hasTrigger: false,
+		},
+		{
+			name: "broker with multiple targets with one matching target filter",
+			brokerTargets: map[string]*config.Target{
+				"non_matching_target_1": {
+					FilterAttributes: map[string]string{
+						"type":   eventType,
+						"source": "some-random-source",
+					},
+				},
+				"non_matching_target_2": {
+					FilterAttributes: map[string]string{
+						"source": "some-random-other-source",
+					},
+				},
+				"matching_target_3": {
+					FilterAttributes: map[string]string{
+						"type": eventType,
+					},
+				},
+			},
+			hasTrigger: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctx := logtest.TestContextWithLogger(t)
+			psSrv := pstest.NewServer()
+			defer psSrv.Close()
+			psClient := createPubsubClient(ctx, t, psSrv)
```

Comment:
`defer psClient.close()`

Reply:
Alternatively, you can use https://github.com/google/knative-gcp/blob/eb5ac1b2f0231ba7bb1681858f3478a001938341/pkg/reconciler/testing/pstest.go#L200